common/src/main/java/com/viaversion/viaversion/rewriter/EntityRewriter.java

```diff
@@ -104,11 +104,10 @@ public void registerFilter(MetaFilter filter) {
     public void handleMetadata(final int entityId, final List<Metadata> metadataList, final UserConnection connection) {
         final TrackedEntity entity = tracker(connection).entity(entityId);
         final EntityType type = entity != null ? entity.entityType() : null;
-        int i = 0; // Count index for fast removal
         for (final Metadata metadata : metadataList.toArray(EMPTY_ARRAY)) { // Copy the list to allow mutation
             // Call handlers implementing the old handleMetadata
             if (!callOldMetaHandler(entityId, type, metadata, metadataList, connection)) {
-                metadataList.remove(i--);
+                metadataList.remove(metadata);
```

Comment:
The toArray call means only the original list that is not being iterated on will be modified. Technically two metadata objects can be equal and result in the wrong one being removed if state is involved, so the easier fix is to just remove the -- in this line

Reply:
The issue is `metadataList` is modified (with remove/add operations) within `callOldMetaHandler` call (removing by object already btw). So the index of iteration here may be different from what the list actually holds.